tests/nix_tests/test_str_v2.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import time
+from web3 import Web3
+
+from .ibc_utils import (
+    EVMOS_IBC_DENOM,
+    assert_ready,
+    get_balance,
+    prepare_network,
+    get_balances,
+)
+from .network import CosmosChain, Evmos
+from .utils import ADDRS, eth_to_bech32, wait_for_fn, erc20_balance
+from .ibc_utils import ATOM_IBC_DENOM
+
+# uatom from cosmoshub-2 -> cosmoshub-1 IBC representation on the Evmos chain.
+ATOM_2_IBC_DENOM_MULTI_HOP = (
+    "ibc/D219F3A490310B65BDC312B5A644B0D56FFF1789D894B902A49FBF9D2F560B32"
+)
+# uatom from cosmoshub-2 -> cosmoshub-1 IBC representation
+ATOM_1_IBC_DENOM_ATOM_2 = (
+    "ibc/C4CFF46FD6DE35CA4CF4CE031E643C8FDC9BA4B99AE598E9B0ED98FE3A2319F9"
+)
+# The ERC20 address of ATOM on Evmos
+ATOM_1_ERC20_ADDRESS = Web3.toChecksumAddress(
+    "0xf36e4C1F926001CEaDa9cA97ea622B25f41e5eB2"
+)
+
+
+@pytest.fixture(scope="module", params=["evmos"])
+def ibc(request, tmp_path_factory):
+    """Prepare the network"""
+    name = "str-v2"
+    evmos_build = request.param
+    path = tmp_path_factory.mktemp(name)
+    # specify the custom_scenario
+    network = prepare_network(path, name, [evmos_build, "cosmoshub-1", "cosmoshub-2"])
+    yield from network
+
+
+def test_str_v2_single_hop(ibc):
+    """
+    Test Single Token Representation v2 with single hop Coin.
+    It should automatically create an ERC20 precompiled contract.
+    And register a token pair.
+    """
+    assert_ready(ibc)
+
+    evmos: Evmos = ibc.chains["evmos"]
+    gaia: CosmosChain = ibc.chains["cosmoshub-1"]
+
+    evmos_cli = evmos.cosmos_cli()
+    evmos_addr = ADDRS["signer2"]
+    bech_dst = eth_to_bech32(evmos_addr)
+
+    gaia_cli = gaia.cosmos_cli()
+    gaia_addr = gaia_cli.address("signer2")
+
+    old_dst_balance = get_balance(evmos, bech_dst, ATOM_IBC_DENOM)
+    rsp = gaia_cli.ibc_transfer(
+        gaia_addr, bech_dst, "5000uatom", "channel-0", 1, fees="10000uatom"
+    )
+    assert rsp["code"] == 0
+
+    # NOTE: Sleep some time because wait_for_fn doesn't work for some reason ?
+    time.sleep(30)
```

Comment:
Did you try here to wait for the IBC Acknoledgement transaction?
One way to do it would be to query the block_results every block and check when the acknoledgement arrives

Reply:
This seems to work nicely for the `evmos_cli` but for some reason not for the cosmos-hub one. Any ideas? Updated the places where it's waiting for the ack.